src/foam/nanos/auth/CachingAuthService.java

```diff
@@ -129,6 +129,8 @@ public void purgeCache(Object obj) {
       userId = ((User) obj).getId();
     } else if ( obj instanceof UserCapabilityJunction ) {
       userId = ((UserCapabilityJunction) obj).getSourceId();
+    } else if (obj instanceof GroupPermissionJunction) {
+      return; //TODO select all user from that group and purge all of them.
```

Comment:
Let's just write the real code and then merge that.
Select all users belonging to this group and then purge their caches.